app/services/slack_service.rb

```diff
@@ -2,7 +2,17 @@ class SlackService
   def initialize(client = Slack::Client.new)
     @client = client
   end
+
   def post_message(channel, message)
     @client.chat_postMessage(channel: channel, text: message, as_user: true)
   end
+
+  def post_direct_message(token, users, message)
+    members = @client.users_list(token: token)
+    user_id_array = []
+    members["members"].each do |key, value|
+      users.each { |user| user_id_array << key["id"] if key["name"] == user }
+    end
+    user_id_array.each { |id| post_message(id, message) }
+  end
```

Comment:
I think we can probably avoid the creation of the `user_id_array` here. Instead of building up the array to then loop through and call `post_message`, I think we can just call `post_message` as soon as we know the `id` for a matching username:

```
  response["members"].each do |member|
      users.each { |user| post_message(member["id"] if member["name"] == user }
  end
```